core/src/main/java/tech/tablesaw/analytic/AggregateFunctions.java

```diff
@@ -213,41 +233,188 @@ public Double getValue() {
-    AggregateFunction<T, Double> slidingFunction() {
-      throw new UnsupportedOperationException("Analytic Function Min Is Not implemented");
+    AggregateFunction<T, Double> functionForSlidingWindows() {
+      return new AggregateFunction<T, Double>() {
+        private final ArrayDeque<Double> queue = new ArrayDeque<>();
+
+        @Override
+        public void removeLeftMost() {
+          queue.remove();
+        }
+
+        @Override
+        public void addRightMost(T newValue) {
+          queue.add(newValue.doubleValue());
+        }
+
+        @Override
+        public void addRightMostMissing() {
+          queue.add(Double.NaN);
+        }
+
+        @Override
+        public Double getValue() {
+          // This could be faster, but probably does not matter in practice because sliding windows
+          // will be small.
+          return queue.stream()
+              .filter(d -> !Double.isNaN(d))
+              .mapToDouble(Number::doubleValue)
+              .min()
+              .orElse(Double.NaN);
+        }
+      };
     }
   }
 
   static class Mean<T extends Number> extends WindowDependentAggregateFunction<T> {
 
     @Override
-    AggregateFunction<T, Double> fixedFunction() {
-      throw new UnsupportedOperationException("Analytic Function Mean Is Not implemented");
+    AppendAggregateFunction<T, Double> functionForAppendWindows() {
+      return new AppendAggregateFunction<T, Double>() {
+        private double sum = Double.NaN;
+        private double count = 0;
+
+        @Override
+        public Double getValue() {
+          if (count == 0) {
+            return Double.NaN;
+          }
+          return sum / count;
+        }
+
+        @Override
+        public void addRightMostMissing() {}
+
+        @Override
+        public void addRightMost(T newValue) {
+          if (Double.isNaN(sum)) {
+            this.sum = 0.0;
+          }
+          this.sum += newValue.doubleValue();
+          count++;
+        }
+      };
     }
 
     @Override
-    AggregateFunction<T, Double> slidingFunction() {
-      throw new UnsupportedOperationException("Analytic Function Mean Is Not implemented");
+    AggregateFunction<T, Double> functionForSlidingWindows() {
+      return new AggregateFunction<T, Double>() {
+        private final ArrayDeque<Double> queue = new ArrayDeque<>();
+        private Double sum = 0.0;
+        private int missingCount = 0;
+
+        @Override
+        public void removeLeftMost() {
+          Double removed = queue.remove();
+          if (Double.isNaN(removed)) {
+            missingCount--;
+          } else {
+            this.sum -= removed;
+          }
+        }
+
+        @Override
+        public void addRightMost(T newValue) {
+          Double doubleValue = newValue.doubleValue();
+          this.sum += doubleValue;
+          queue.add(doubleValue);
+        }
+
+        @Override
+        public void addRightMostMissing() {
+          queue.add(Double.NaN);
```

Comment:
should we use the missingValueIndicator functions in case an implementation changes

Reply:
Good point. Done